projects/telemetry/src/gps.c

```diff
@@ -0,0 +1,78 @@
+#include "gps.h"
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include "delay.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "uart.h"
+#include "xbee.h"
```

Comment:
Is this coupling between the GPS and Xbee really necessary?

You can change your API to provide a way to register a callback, and then register `xbee_transmit` through that, which breaks that dependency.